### PR TITLE
Update batchnormalization to PyTorch 1.0

### PR DESCRIPTION
**Pull request by jatentaki**
* Fix breakage with pytorch 1.0
* Factor out batchnorm.py tests into a standalone file, add more tests, use `unittest` lib

### CODE REVIEW

**Comment by mariogeiger** [comment]
Thanks

Two questions:
1) I don't what is better, to import like `from batchnorm import SE3BNConvolution` or `from se3cnn import SE3BNConvolution` ?
2) I know nothing about `unittest`. Should we create a `tests` folder with all the tests ?

I merge this PR now.

**Comment by jatentaki** [comment]
1. It's a good question. I went for the former because it doesn't force me to reinstall the module with `setup.py` each time I want to run the suite, which I think avoids a potential footgun when you're testing and (not) seeing an error, when it's just your installation being outdated. So I think for dev it's better. Then for distributing, you probably want to let people run the tests and verify their installation and not the installer...
2. I don't know much either, I just googled it and used the first unit testing lib that came up. I wanted something which lets me see all the tests instead of failing on the first throw (so I can figure out the most likely culprit looking at what passed and what did not). I think a standalone tests directory would is a good good idea, I did it in a quick-and-dirty fashion here.

I have a bunch of new tests and fixes ready, which enable `torch.jit` and `torch.nn.DataParallel` to be used alongside the `SE3` modules. Coming tomorrow

File: se3cnn/bnorm_tests.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            # rotate 90 degrees in plane of axes 2 and 3
+            return torch.flip(t, (2, )).transpose(2, 3)
+
+        def unrotate(t):
+            # undo the rotation by 3 more rotations
+            return rotate(rotate(rotate(t)))
+
+        inp = torch.randn(4, 1, 12, 12, 12)
+
+        if cuda:
```
Comment (mariogeiger) [inline se3cnn/bnorm_tests.py:43]: Isn't it better tu use torch.device ?

File: se3cnn/bnorm_tests.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    def test_se3bnconv_equivariance_cpu(self):
+        self._test_equivariance(module=SE3BNConvolution, cuda=False)
+
+    def test_se3bn_equivariance_cuda(self):
+        self._test_equivariance(module=JointConvolution, cuda=True)
+
+    def test_se3bn_equivariance_cpu(self):
+        self._test_equivariance(module=JointConvolution, cuda=False)
+
+    def test_does_reduce_variance(self):
```
Comment (mariogeiger) [inline se3cnn/bnorm_tests.py:70]: The name is weird.
This test check that `SE3BNConvolution(x) = SE3Convolution(SE3BatchNorm(x))`.
It does not look if the variance changes.

**Comment by mariogeiger** [comment]
1. It makes sense, you could also avoid reinstall with `python setup.py develop` instead of `python setup.py install`

> I have a bunch of new tests and fixes ready, which enable `torch.jit` and `torch.nn.DataParallel` to be used alongside the `SE3` modules. Coming tomorrow

Very cool, I cant wait to see that!
